New Mini-Orchestration Module - hubPeeredSpoke (#223)
Azure/ALZ-Bicep: infra-as-code/bicep/orchestration/hubPeeredSpoke/README.md

> # Module: orchestration\hubPeeredSpoke - Spoke network, including peering to Hub

Reviewer: ```suggestion
# Module: Orchestration - hubPeeredSpoke - Spoke network, including peering to Hub (Hub & Spoke or Virtual WAN)
```

Author: Updated in newest commit

> This module acts as an orchestration module that create and configures a spoke network to deliver the Azure Landing Zone Hub & Spoke architecture which is also described in the wiki on the [Deployment Flow article](https://github.com/Azure/ALZ-Bicep/wiki/DeploymentFlow).

Reviewer: ```suggestion
This module acts as an orchestration module that create and configures a spoke network to deliver the Azure Landing Zone Hub & Spoke architecture, for both traditional Hub & Spoke and Virtual WAN, which is also described in the wiki on the [Deployment Flow article](https://github.com/Azure/ALZ-Bicep/wiki/DeploymentFlow).
```

Module deploys the following resources:

- Subscription placement in Management Group hierarchy - if parPeeredVnetSubscriptionMGPlacement is specified
- Virtual Network (Spoke VNet)
- UDR - if parNextHopIPAddress and resource id of hub virtual network object is specified
- Hub to Spoke peering - if resource id of hub virtual network object is specified in parHubVirtualNetworkID
- Spoke to hub peering - if resource id of hub virtual network object is specified in parHubVirtualNetworkID
- Spoke to virtual WAN peering - if resource id of virtual WAN hub object is specified in parHubVirtualNetworkID

> 

Reviewer: Can we add a note here that you can only choose the traditional Hub & Spoke method OR VWAN and not both?

Just to avoid confusion where someone tries to do traditional peering and also VWAN connection to a hub?

Author: Updated in newest commit

- Spoke to hub peering - if resource id of hub virtual network object is specified in parHubVirtualNetworkID
- Spoke to virtual WAN peering - if resource id of virtual WAN hub object is specified in parHubVirtualNetworkID

Note that only one peering type can be created with this module, so either traditional Hub & Spoke OR Azure virtual WAN.

## Parameters

The module requires the following inputs:

 | Parameter                              | Type   | Default                                                         | Description                                                            | Requirement        | Example                                                                                                                                                    |
 | -------------------------------------- | ------ | --------------------------------------------------------------- | ---------------------------------------------------------------------- | ------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |

>  | parLocation                            | string | Empty string `''`                                               | The region to deploy all resoruces into                                | Valid Azure Region | `northeurope`                                                                                                                                              |

Reviewer: ```suggestion
 | parLocation                            | string | Empty string `''`                                               | The region to deploy all resources into                                | Valid Azure Region | `deployment().location`                                                                                                                                              |
```

Author: Updated in ThirdCommit
